app/services/map_data_service.py

```diff
@@ -0,0 +1,288 @@
+            return None
+
+    def get_formatted_address(self, place_id: str) -> str:
+        try:
+            place = self.client.place(place_id=place_id)
+            return place.get("result", {}).get("formatted_address", "")
+        except Exception as e:
+            print(f"❌ Error: {e}")
+            return None
+
+    def get_phone_number(self, place_id: str) -> str:
+        try:
+            place = self.client.place(place_id=place_id)
+            return place.get("result", {}).get("international_phone_number", "")
+        except Exception as e:
+            print(f"❌ Error: {e}")
+            return None
+
+    def get_opening_hours(self, current_opening_hours: Dict[str, Any]) -> OpeningHours:
+        today = datetime.now()
+        week_start = today - timedelta(days=today.weekday())
+        week_end = week_start + timedelta(days=6)
+        
+        week_start_str = week_start.strftime("%Y-%m-%d")
+        week_end_str = week_end.strftime("%Y-%m-%d")
+        last_refreshed = week_end.replace(hour=21, minute=0, second=0).isoformat() + "+03:00"
+        
+        weekly = {}
+        weekday_names = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"]
+        
+        periods = current_opening_hours.get("periods", [])
+        
+        for period in periods:
+            open_info = period.get("open", {})
+            close_info = period.get("close", {})
+            
+            day_name = weekday_names[open_info.get("day", 0)]
+            start_time = open_info.get("time", "12:00")
+            end_time = close_info.get("time", "21:00")
+            
+            start_formatted = f"{start_time[:2]}:{start_time[2:]}"
+            end_formatted = f"{end_time[:2]}:{end_time[2:]}"
+            
+            if day_name not in weekly:
+                weekly[day_name] = []
+            
+            weekly[day_name].append({
+                "start": start_formatted,
+                "end": end_formatted
+            })
+        
+        for day in weekday_names:
+            if day not in weekly:
+                if day in ["Saturday", "Sunday"]:
+                    weekly[day] = [{"start": "10:00", "end": "21:00"}]
+                else:
+                    weekly[day] = [{"start": "12:00", "end": "21:00"}]
+        
+        return OpeningHours(
+            type="weekly",
+            week_start=week_start_str,
+            week_end=week_end_str,
+            last_refreshed=last_refreshed,
+            weekly=weekly
+        )
+
+    def get_place_id(self, place: str) -> str:
+        try:
+            return self.client.find_place(place, "textquery")
+        except Exception as e:
+            print(f"❌ Error: {e}")
+            return None
+
+    def get_places_ids(self, places: List[str]) -> List[str]:
+        try:
+            place_ids = []
+            for place in places:
+                place_ids.append(self.client.find_place(place, "textquery"))
+            return place_ids
```

Comment:
Same as above: this method may not return a list of `place_id` strings as expected. Extract the relevant field.